File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.kt
class Solution {
    fun solution(n: Int, a: Int, b: Int): Int {
        var answer = 0
        var _a = a
        var _b = b
        
        while (_a != _b) {
            ++answer
            _a = next_round(_a)
            _b = next_round(_b)
        }

        return answer
    }
    
    fun next_round(num: Int,): Int {
        return (num+1)/2
    }
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    answer = 0
    while True:
        # if (a+1 == b or b+1 == a) and (min(a, b)%2 == 1):
        if (min(a, b)+1 == max(a, b)) and (min(a, b)%2 == 1):
            answer += 1
            break
        a = next_round(a)
        b = next_round(b)
        answer += 1
        
    return answer

def next_round(a):
    if a%2==0:
        a //= 2
    else:
        a = a//2 + a%2

    return a
# 홀수 -> 1이면 1, 3이면 2, 5이면 3
# 짝수 -> 2이면 1, 4이면 2, 6이면 3
# -> 1 또는 2가 될때 까지 반복
